src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -112,7 +113,13 @@ protected override Task UpdateDisplayAsync(in RenderBatch batch)
             // snapshot its contents now.
             // TODO: Consider using some kind of array pool instead of allocating a new
             //       buffer on every render.
-            var batchBytes = MessagePackSerializer.Serialize(batch, RenderBatchFormatterResolver.Instance);
+            var memoryStream = new MemoryStream();
+            using (var renderBatchWriter = new RenderBatchWriter(memoryStream, false))
```

Comment:
Hope there are some perf improvement possibilities here :D

Reply:
There's a TODO there which we'll get to at some point. Right now the focus is to get something in.